benchmarks/polys.py

```diff
@@ -1,17 +1,146 @@
-import sympy
+from sympy import symbols, prod
+from sympy.polys import ZZ, Poly
+
 
 class TimePolyManyGens:
     """Time using a Poly with many generators"""
 
     params = [1, 10, 100, 500]
 
     def setup(self, n):
-        self.xs = sympy.symbols('x:{}'.format(n))
+        self.xs = symbols('x:{}'.format(n))
         self.x = self.xs[n // 2]
-        self.px = sympy.Poly(self.x, self.xs)
+        self.px = Poly(self.x, self.xs)
 
     def time_create_poly(self, n):
-        sympy.Poly(self.x, self.xs)
+        Poly(self.x, self.xs)
 
     def time_is_linear(self, n):
         self.px.is_linear
+
+
+
+x, y, y1, y2, y3, y4, y5, y6, y7, y8, y9, y10 = symbols("x y y1 y2 y3 y4 y5 y6 y7 y8 y9 y10")
+
+y = [y1, y2, y3, y4, y5, y6, y7, y8, y9, y10] # set values of y for v = 1 to 10
+
+R = ZZ[x, y, y1, y2, y3, y4, y5, y6, y7, y8, y9, y10]
+
+# Case: Linearly dense quartic inputs with quadratic GCDs
+def bench1_core_prem():
+    for v in range(1, 11):
+        D = (1 + x + sum(y[:v])) ** 2
+        f = D * (-2 + x - sum(y[:v])) ** 2
+        g = D * (2 + x + sum(y[:v])) ** 2
+
+        return prem(f, g, x)
```

Comment:
We should time separately for each value of v. This also includes the time to construct the polynomials which it should not.

The benchmarks should be more like this one:
https://github.com/sympy/sympy_benchmarks/blob/3680cc194e5af27663c7e46215bf89789766291e/benchmarks/integrate.py#L46-L66